### PR TITLE
[#34] add Gravitino Spark connector to playground

### PR DESCRIPTION
**Pull request by FANNG1**
add spark connector to playground

fixes: #34

### CODE REVIEW

File: init/hive/init.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 sed -i '$d' /usr/local/sbin/start.sh
 sed -i '$d' /usr/local/sbin/start.sh
+sed -i 's|hdfs://localhost:9000|hdfs://hive:9000|g' /usr/local/hive/conf/hive-site.xml
```
Comment (FANNG1) [inline init/hive/init.sh:8]: spark sql create database will use `hive.metastore.warehouse.dir` in hive-site.xml, we should correct the hdfs address

File: init/spark/init.sh [hunk 2]
```diff
@@ -4,5 +4,8 @@
 #
 mkdir -p /opt/spark/conf
 cp /tmp/spark/spark-defaults.conf /opt/spark/conf
-wget https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.4_2.12/1.3.1/iceberg-spark-runtime-3.4_2.12-1.3.1.jar  -O /opt/spark/jars/iceberg-spark-runtime-3.4_2.12-1.3.1.jar
```
Comment (FANNG1) [inline init/spark/init.sh:7]: 1.3.1 doesn't support Iceberg JDBC catalog

File: init/spark/spark-defaults.conf [hunk 3]
```diff
@@ -3,8 +3,13 @@
 # This software is licensed under the Apache License version 2.
 #
 
+spark.plugins com.datastrato.gravitino.spark.connector.plugin.GravitinoSparkPlugin
+spark.sql.gravitino.uri http://gravitino:8090
+spark.sql.gravitino.metalake metalake_demo
+spark.sql.gravitino.enableIcebergSupport true 
 spark.sql.extensions org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions
-spark.sql.catalog.catalog_iceberg org.apache.iceberg.spark.SparkCatalog
```
Comment (FANNG1) [inline init/spark/spark-defaults.conf:7]: duplicate catalog name with Gravitinio `catalog_iceberg`

**Comment by FANNG1** [comment]
@qqqttt123 , could you help to review when you free?

**Comment by jerqi** [comment]
There are some issues.
1. Trino and Spark will init the metalake twice.
2. Is there the conflict if we want to use Spark rest catalog and Spark connector at the same time?
3. Do you need a case about Spark connector?

**Comment by FANNG1** [comment]
> There are some issues.
> 
> 1. Trino and Spark will init the metalake twice.

the script checks the exist of metalake and catalogs before creating it.

> 2. Is there the conflict if we want to use Spark rest catalog and Spark connector at the same time?

no, I changed the rest catalog name to avoid conflict.

> 3. Do you need a case about Spark connector?

no, it's out of the scope of this PR

**Comment by jerqi** [comment]
You should include the case. Because it's the verification about the Spark connector.

**Comment by FANNG1** [comment]
> You should include the case. Because it's the verification about the Spark connector.

this pr only focus on add spark connector to playground, we could use another pr to do it

**Comment by jerqi** [comment]
> > You should include the case. Because it's the verification about the Spark connector.
> 
> this pr only focus on add spark connector to playground, we could use another pr to do it

You should add the document or you add the steps of your test at least. Otherwise, I don't how to verify this pull request.

**Comment by FANNG1** [comment]
> > > You should include the case. Because it's the verification about the Spark connector.
> > 
> > 
> > this pr only focus on add spark connector to playground, we could use another pr to do it
> 
> You should add the document or you add the steps of your test at least. Otherwise, I don't how to verify this pull request.

so, I propose another PR in Gravitino playground document?

**Comment by FANNG1** [comment]
@jerqi , update the document, please help to review again

File: init/spark/init.sh (hunk 2 above)
Comment (jerqi) [reply init/spark/init.sh:21]: Maybe we should merge this pull request after we release the 0.6.0.

Comment (FANNG1) [reply init/spark/init.sh:21]: ok, any other comments?

**Comment by jerqi** [comment]
Will it influence the demo of Iceberg REST catalog?
Is it clear for user to use Spark connector and REST service catalog. 
It is a confusing for user.

**Comment by jerqi** [comment]
You should submit the document to the main repo, too.

**Comment by FANNG1** [comment]
> Will it influence the demo of Iceberg REST catalog? Is it clear for user to use Spark connector and REST service catalog. It is a confusing for user.

The original document makes Spark and Trino use same catalog name `iceberg_catalog`, but they are different, maybe we could add some comment to tell user `iceberg_catalog` and `rest_catalog` refer to the same underlying catalog, WDYT?

**Review by jerqi — APPROVED** [review]
LGTM.

File: init/gravitino/gravitino.conf [hunk 4]
```diff
@@ -67,3 +67,4 @@ gravitino.auxService.iceberg-rest.warehouse = hdfs://hive:9000/user/iceberg/ware
 gravitino.auxService.iceberg-rest.jdbc.user = mysql
 gravitino.auxService.iceberg-rest.jdbc.password = mysql
 gravitino.auxService.iceberg-rest.jdbc-driver = com.mysql.cj.jdbc.Driver
+gravitino.auxService.iceberg-rest.catalog-backend-name = catalog_iceberg
```
Comment (FANNG1) [inline init/gravitino/gravitino.conf:70]: let REST catalog use the same catalog name with Gravitino Iceberg catalog, so they could query same table. cc @jerqi

**Comment by FANNG1** [comment]
@jerqi , I have tested spark trino and Iceberg REST service part, please help to review again

File: README.md [hunk 5]
```diff
@@ -75,17 +75,31 @@ docker exec -it playground-trino bash
 trino@container_id:/$ trino
 ```
 
-### Using Jupiter Notebook
+## Using Jupiter Notebook
```
Comment (jerqi) [inline README.md:78]: Jupyter

Comment (FANNG1) [reply README.md]: done

**Review by jerqi — APPROVED** [review]
M

**Review by yuqi1129 — APPROVED** [review]

**Pull request merged**